src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt

```diff
@@ -209,4 +225,11 @@ class AddressService(
       isMainResident = this.isMainResident,
       standardAddressCheckRequest = this.standardAddressCheckRequest.toSummary(),
     )
+
```

Comment:
I think there must be a better way of doing this because it is doing this.toSummary for both types. But I don't really want to add toSummary to the entity abstract class as that is giving it knowledge about layer above.

Reply:
can you make it sealed to remove the `else` ? I'm guessing probably not.

Reply:
I will get back to you in a bit on that, it does appear to compile, I try running the integration tests

Reply:
A shame unit tests were ok, but hibernate doesn't like it

  ```
........
  Caused by: org.hibernate.MappingException: Could not instantiate persister org.hibernate.persister.entity.JoinedSubclassEntityPersister

    Caused by: java.lang.IllegalArgumentException: Could not create type

    Caused by: java.lang.IncompatibleClassChangeError: class uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.CurfewAddressCheckRequest$HibernateProxy$vhELgVIP cannot inherit from sealed class uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.CurfewAddressCheckRequest
```

Reply:
👍 think it's fine as is